### PR TITLE
Add type checking of input to main function

### LINKED ISSUE #4: General sanity checks on inputs
There are no test files for handling edge cases like 0x0 matrices or invalid inputs like lists of lists which nevertheless are matrices. For this there should be both test files and ofc code that handles these cases.

### PR DESCRIPTION
**Pull request by emmakatherina**
Fixes #4. Please review.

What other checks should be done?
I thought about checking whether the height and width of the matrix matches the values of a and b. But why do we need a and b if we could check this by ourselves?

### CODE REVIEW

**Comment by ssiccha** [comment]
> I thought about checking whether the height and width of the matrix matches the values of a and b. But why do we need a and b if we could check this by ourselves?

Because a and b are not the dimensions of the matrix but the number of chops in horizontal and vertical direction.

File: main.g [hunk 1]
```diff
@@ -54,14 +54,28 @@ Chief := function( galoisField,mat,a,b,IsHPC )
     ##Preparation: Init and chopping the matrix mat
     Info(InfoGauss, 1, "------------ Start Chief ------------");
     Info(InfoGauss, 1, "Preparation");
+    
+    Info(InfoGauss, 3, "Input checks");
+    if not IsField(galoisField) then
```
[the remark this answers is not in the document]
Comment (emmakatherina) [reply main.g]: Thank you! I will try to think about that in the future.
